httpcore5/src/main/java/org/apache/hc/core5/http/message/BufferedHeader.java

```diff
@@ -110,7 +113,27 @@ public String getName() {
 
     @Override
     public String getValue() {
-        return this.buffer.substringTrimmed(this.valuePos, this.buffer.length());
+        if (value == null) {
+            int beginIdx = valuePos;
+            int endIdx = buffer.length();
+            while (beginIdx < buffer.length() && Tokenizer.isWhitespace(buffer.charAt(beginIdx))) {
+                beginIdx++;
+            }
+            while (endIdx > beginIdx && Tokenizer.isWhitespace(buffer.charAt(endIdx - 1))) {
+                endIdx--;
+            }
+            final StringBuilder buf = new StringBuilder(endIdx - beginIdx);
+            for (int i = beginIdx; i < endIdx; i++) {
+                final char ch = buffer.charAt(i);
+                if (ch == Chars.CR || ch == Chars.LF || ch == 0) {
+                    buf.append(' ');
```

Comment:
Using `(char) Chars.SP` instead of ' ' should not degrade the performance and would be clearer.

Reply:
@arturobernalg Done.